### PR TITLE
1143-When-parsing-Smalltalk-source-code-methods-stubs-are-not-created

### PR DESCRIPTION
**Pull request by NicolasAnquetil**
First crude implementation of stub method importation in Pharo
The creation is not done by default so it does not change the usual behaviour.

### CODE REVIEW

**Comment by Ducasse** [comment]
Nicolas to help cross referencing it is nice to add #number of issue like that a link is created and we just have to click.

**Comment by Ducasse** [comment]
stub methods where not created?
when do you create them? when there is a message and the method is not found?

**Comment by Ducasse** [comment]
I read the code and it looks ok modulo the little duplication (but fixing it is not easy since the body extraction is inside a loop). Now I have a question what is the class of a stub method. 
I have the impression that it would make sure to have a stub class a bit like a null pattern?
Or a stub method should be from a specific class because if you print a method
class >> #selector then the printing behavior of normal method will have to check if the method is a stub and avoid to print the class part.

**Comment by NicolasAnquetil** [comment]
Yes, this is a rough implementation:
- Parent class should be created as a stub
- I would also like to look at doing the InstallElementsOperator _after_ the CandidateListOperator (in `MooseCompositeImporterTask>>basicRun`)
- see if the new option (shouldImportSmalltalkStubMethods) can be put in FamixStImportingContext
- testing the stuff for real

File: src/Moose-SmalltalkImporter/CandidateListOperator.class.st [hunk 1]
```diff
@@ -13,12 +13,15 @@ Class {
...
 { #category : #running }
 CandidateListOperator >> allCandidatesFor: invocation [ 
-	 
 	^Array 
 		withAll: 
 			(groups 
 				at: invocation signature 
-				ifAbsent: #())
+				ifAbsent: [ | stubs |
+					stubs := self importer ensureSmalltalkStubMethodsFor: invocation signature.
```
Comment (jecisc) [inline src/Moose-SmalltalkImporter/CandidateListOperator.class.st:21]: Should this be here or in AbstractCandidateListOperator?

Comment (NicolasAnquetil) [reply src/Moose-SmalltalkImporter/CandidateListOperator.class.st:21]: There must be something in the #ifAbsent: and what was there before was not good.

There are other places were the whole solution could be improved, but currently it works and does not break anything.
We will make it better some other time (perfect is the enemy of good)

Comment (NicolasAnquetil) [reply src/Moose-SmalltalkImporter/CandidateListOperator.class.st:21]: OK, I hadn't understood the problem with putting it here or in the super class.
At first glance, I would say you are right and it should be in the super-class.
But I would have to spend more time on it to be sure.

**Comment by jecisc** [comment]
@NicolasAnquetil Is it ready for merge?

**Pull request merged**